### DISCUSSION TITLE
Self contained deckset

### DISCUSSION THREAD

**Discussion by ddddavidee** [marp-team/marp#132] [Q&A] [answered]
Is it possible to create a self contained presentation to use offline? 
it could be with all javascript and fonts embedded in the html itself or also in a folder distributed with the presentation itself.

(I would like to present something on an offline pc and I use quite a lot the KaTeX extension)

thanks in advance, d

**Comment by yhatt** [comment, marked as answer]
Marp Community often says using Marp in combination with [monolith](https://github.com/Y2Z/monolith) (e.g. https://github.com/marp-team/marp-cli/issues/302, https://github.com/marp-team/marp/discussions/122#discussioncomment-775560). It bundles the outputted web page into a single HTML file.

```shell
$ marp -o - markdown.md | monolith -o bundled.html -
```

And please remember that [PDF export is an easier way for your purpose](https://marp.app/docs#export-to-pdf-pptx-html).

**Reply by ddddavidee** [reply]
Thanks a lot for your reply. 
Before asking I tried with a similar solution (the firefox extension: https://addons.mozilla.org/en-US/firefox/addon/single-file/) but it only saved the current page.

monolith works very well, thanks a lot.

I known about the pdf export, but I've some gif that cannot be exported as animation (as far as I know)
